src/arch/armv8/aarch32/inc/arch/subarch/sysregs.h

```diff
@@ -38,7 +38,8 @@
 #define SYSREG_GEN_ACCESSORS_64(reg, op1, crm)                                                    \
     static inline unsigned long long sysreg_##reg##_read(void)                                    \
     {                                                                                             \
-        unsigned long long _temp, _tempH;                                                         \
+        unsigned long long _tempH;                                                                \
+        unsigned long _temp;                                                                      \
         __asm__ volatile("mrrc p15, " #op1 ", %0, %1, " #crm "\n\r" : "=r"(_temp), "=r"(_tempH)); \
         return ((_tempH << 32) | _temp);                                                          \
```

Comment:
I see the problem. But maybe we can add the explicit cast at the end:

```suggestion
        return ((_tempH << 32) | ((unsigned long) _temp));                                                          \
```

Reply:
Yes, that also works! I'll change to your suggestion.
Thanks!